internal/controller/finbackupconfig_controller.go

```diff
@@ -52,5 +102,158 @@ func (r *FinBackupConfigReconciler) Reconcile(ctx context.Context, req ctrl.Requ
+		return "", nil
+	}
+
+	clusterID, err := getCephClusterIDFromSCName(ctx, k8sClient, *storageClassName)
+	if err != nil {
+		logger.Info("failed to get ceph cluster ID from StorageClass name",
+			"error", err, "namespace", pvc.Namespace, "pvc", pvc.Name, "storageClassName", *storageClassName)
+		if errors.Is(err, errEmptyClusterID) {
+			return "", nil
+		}
+		return "", err
+	}
+
+	return clusterID, nil
+}
+
+func getRunningPod(ctx context.Context, client client.Client) (*corev1.Pod, error) {
+	name, ok := os.LookupEnv("POD_NAME")
+	if !ok {
+		return nil, fmt.Errorf("POD_NAME not found")
+	}
+	namespace, ok := os.LookupEnv("POD_NAMESPACE")
+	if !ok {
+		return nil, fmt.Errorf("POD_NAMESPACE not found")
+	}
+	var pod corev1.Pod
+	if err := client.Get(ctx, types.NamespacedName{Name: name, Namespace: namespace}, &pod); err != nil {
+		return nil, fmt.Errorf("failed to get pod: %w", err)
+	}
+	return &pod, nil
+}
+
+func setEnvFromFieldRef(container *corev1.Container, envName, fieldPath string) {
+	envIndex := slices.IndexFunc(container.Env, func(e corev1.EnvVar) bool {
+		return e.Name == envName
+	})
+	if envIndex == -1 {
+		container.Env = append(container.Env, corev1.EnvVar{Name: envName})
+		envIndex = len(container.Env) - 1
+	}
+	env := &container.Env[envIndex]
+	if env.ValueFrom == nil {
+		env.ValueFrom = &corev1.EnvVarSource{}
+	}
+	if env.ValueFrom.FieldRef == nil {
+		env.ValueFrom.FieldRef = &corev1.ObjectFieldSelector{}
+	}
+	env.ValueFrom.FieldRef.FieldPath = fieldPath
+}
+
+func (r *FinBackupConfigReconciler) createOrUpdateCronJob(
+	ctx context.Context,
+	fbc *finv1.FinBackupConfig,
+	namespace string,
+	serviceAccountName string,
+	image string,
+) error {
+	logger := log.FromContext(ctx)
+	cronJobName := "fbc-" + string(fbc.UID)
+
+	cronJob := &batchv1.CronJob{}
+	cronJob.SetName(cronJobName)
+	cronJob.SetNamespace(namespace)
+
+	op, err := ctrl.CreateOrUpdate(ctx, r.Client, cronJob, func() error {
+		cronJob.Spec.Schedule = fbc.Spec.Schedule
+		if r.overwriteFBCSchedule != "" {
+			cronJob.Spec.Schedule = r.overwriteFBCSchedule
+		}
+		cronJob.Spec.Suspend = &fbc.Spec.Suspend
+		var startingDeadlineSeconds int64 = 3600
+		cronJob.Spec.StartingDeadlineSeconds = &startingDeadlineSeconds
+		cronJob.Spec.ConcurrencyPolicy = batchv1.ForbidConcurrent
+		var backoffLimit int32 = 65535
+		cronJob.Spec.JobTemplate.Spec.BackoffLimit = &backoffLimit
+
+		// Set Job creation timestamp as RFC3339 string into PodTemplate annotations
+		creationTime := metav1.Now().Format(time.RFC3339)
+		if cronJob.Spec.JobTemplate.Annotations == nil {
+			cronJob.Spec.JobTemplate.Annotations = make(map[string]string)
+		}
+		cronJob.Spec.JobTemplate.Annotations["job.k8s.io/created-at"] = creationTime
+		if cronJob.Spec.JobTemplate.Spec.Template.Annotations == nil {
+			cronJob.Spec.JobTemplate.Spec.Template.Annotations = make(map[string]string)
+		}
+		cronJob.Spec.JobTemplate.Spec.Template.Annotations["job.k8s.io/created-at"] = creationTime
```

Comment:
I don't think the `created-at` annotation used here will work correctly. The call of `metav1.Now()` returns not the time when a Job is created from the CronJob but the time when the CronJob is created or updated, which is not what we want.

I think it's infeasible to pass the Job creation time through environment variables (i.e., our detailed specification is actually wrong). One possible way is to get the Job resource in the create-finbackup-job subcommand and inspect its `.metadata.creationTimestamp`, but I'm not sure it's the best one.